[Desean Soh] iP

deseansoh opened the pull request:
# ThoughtBot
> Thought about it? ThoughtBot it.

:rocket: **ThoughtBot** is the all-new CLI-based task management chatbot for all your tracking needs. It currently has the following features:
- :grey_question: Helpful replies with suggestions on typos
- :pencil2: Easy to pick up
- :memo: Memory between multiple sessions

Just follow the below steps to use it right now!
1. Download the release file from [here](https://github.com/deseansoh/ip/releases/tag/A-Jar)
2. Launch the app
3. Explore and use ThoughtBot 👍 

Current _planned features_ include:
- [x] 'Find' feature: Finding tasks based on their task descriptions
- [x] Adding a GUI using JavaFX

The code is fully open source as well: the main logic is handled in the `Parser`, `Ui`, and `TaskList` classes. Feel free to fork the project and personalise your very own ThoughtBot!
```java
public static void main(String[] args) {
    if (interested) {
        download();
    } else {
        sad();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

deseansoh committed:
Rename bot

deseansoh committed:
Add greeting and goodbye

deseansoh committed:
Add Scanner library

deseansoh committed:
Add userInput and userInputScanner variables

deseansoh committed:
Add echo logic

deseansoh committed:
Abstract getUserInput behaviour

deseansoh committed:
Separate Constants into new class

deseansoh committed:
Add INDENT constant

deseansoh committed:
Add Entry class

deseansoh committed:
Add basic TaskList class

deseansoh committed:
Add methods addEntry and displayTaskList to TaskList

deseansoh committed:
Add add and list logic to main

deseansoh committed:
Abstract greeting and goodbye

deseansoh committed:
Add done functionality to Entry

deseansoh committed:
Add Command enum

deseansoh committed:
Add UserCommand and child classes

deseansoh committed:
Add Parser

deseansoh committed:
Add mark function

deseansoh committed:
Refactor Constants to StringConstants

deseansoh committed:
Edit variable names in ThoughtBot and Parser

deseansoh committed:
Move println functions from TaskList to ThoughtBot

deseansoh committed:
Rename markUndone to unmarkDone

deseansoh committed:
Add unmark logic in ThoughtBot

deseansoh committed:
Add unmarkEntry method to TaskList, remove indents

deseansoh committed:
Abstract indenting and formatting responses

deseansoh committed:
Make Entry abstract class, add EntryToDo

deseansoh committed:
Override getName method in EntryToDo

deseansoh committed:
Add UserCommandTodo

deseansoh committed:
Edit logic to work with UserCommandTodo

deseansoh committed:
Add UserCommandDeadline

deseansoh committed:
Add EntryDeadline

deseansoh committed:
Add deadline logic to rest of code

deseansoh committed:
Add UserCommandEvent

deseansoh committed:
Add EntryEvent

deseansoh committed:
Add event logic to rest of code

deseansoh committed:
Add basic TextUI testing

deseansoh committed:
Add UserCommandError

deseansoh committed:
Add getTaskNumbersString to TaskList

deseansoh committed:
Add error handling for todo

deseansoh committed:
Rework error message to use exceptions

deseansoh committed:
Rework unrecognised command to be exception

deseansoh committed:
Add UnrecognisedKeywordException

deseansoh committed:
Add error handling for deadline

deseansoh committed:
Add CommandFormats

deseansoh committed:
User CommandFormats in Parser

deseansoh committed:
Add error handling for event

deseansoh committed:
Add error handling for mark/unmark

deseansoh committed:
Refactor Entry to Task

deseansoh committed:
Add OutOfBoundsTaskException

deseansoh committed:
Add text-ui-test/EXPECTED-ERRORHANDLING-UNIX.TXT to gitignore

deseansoh committed:
Add tests for error handling

deseansoh committed:
Add UserCommandDelete

deseansoh committed:
Add delete functionality

deseansoh committed:
Add error handling for delete

deseansoh committed:
Remove wildcard import statements

deseansoh committed:
Remove redundant UserCommandError class

deseansoh committed:
Add comments to all classes

deseansoh committed:
Rename package userCommands to usercommands

deseansoh committed:
Add thoughtbot package

deseansoh committed:
Refactor tasks getName functions

deseansoh committed:
Create SaveLoad class, save function

deseansoh committed:
Add savefile to gitignore

deseansoh committed:
Add LoadErrorException

deseansoh committed:
Finish SaveLoad class

deseansoh committed:
Merge branch 'branch-level-7'

deseansoh committed:
Change deadline command format

deseansoh committed:
Change UserCommandDeadline deadline type

deseansoh committed:
Change TaskDeadline deadline type and getDeadline

deseansoh committed:
Change TaskList addDeadline type

deseansoh committed:
Add comments for Deadline tasks

deseansoh committed:
Add DateTimeFormatException

deseansoh committed:
Edit UnrecognisedCmdException string

deseansoh committed:
Add deadline type and exceptions to Parser and SaveLoad

deseansoh committed:
Change event command format

deseansoh committed:
Change UserCommandEvent fromTime and toTime type

deseansoh committed:
Change TaskEvent fromTIme and toTime type

deseansoh committed:
Change getFromTime and getToTime

deseansoh committed:
Change TaskList addEvent type

deseansoh committed:
Change UserCommandEvent type

deseansoh committed:
Add event changes to Parser and SaveLoad

deseansoh committed:
Merge branch 'branch-level-8'

deseansoh committed:
Add Ui class

deseansoh committed:
Rename boolean variables to follow style

deseansoh committed:
Merge branch 'add-gradle-support'

deseansoh committed:
Add config files for checkstyle

deseansoh committed:
Editing codebase to follow checkstyle

deseansoh committed:
Add tests for tasks package

deseansoh committed:
Add tests for list, todo, and deadline for Parser class

deseansoh committed:
Update shadowJar properties in build.gradle

deseansoh committed:
Fix typo in JavaDoc

deseansoh committed:
Add comments to main method

deseansoh committed:
Add UserCommandFind and Command.Find

deseansoh committed:
Add command format for find

deseansoh committed:
Add find logic to parser

deseansoh committed:
Add TaskNotFoundException

deseansoh committed:
Add logic for find to TaskList and Ui

deseansoh committed:
Merge branch 'branch-A-JavaDoc'

deseansoh committed:
Merge branch 'branch-A-CodingStandard'

deseansoh committed:
Merge branch 'branch-Level-9'

deseansoh committed:
Add comments to findTask in TaskList

deseansoh committed:
Import javafx-tutorial code

deseansoh committed:
Edit codebase to fit javafx

deseansoh committed:
Edit codebase to follow checkstyle

deseansoh committed:
Add branch for tracking

deseansoh committed:
Merge branch 'branch-Level-10'

shuang26 commented on src/main/java/thoughtbot/SaveLoad.java:
```diff
@@ -0,0 +1,178 @@
+     * @param t Task to be parsed
+     * @return String format of the task for the save file
+     */
+    private static String parseToSaveFormat(Task t) {
```
method name is clear and easy to understand 👍

shuang26 commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,66 @@
+ */
+public abstract class Task {
+    private String name;
+    private boolean isDone;
```
variables are in camel case. well done

shuang26 commented on src/main/java/usercommands/UserCommand.java:
```diff
@@ -0,0 +1,20 @@
+ * the user input. Every UserCommand object will have a commandType variable that encapsulates
+ * the command the user is trying to execute.
+ */
+public abstract class UserCommand {
```
class name is easy to understand

shuang26 reviewed (commented):
Overall, the naming for classes, variables and methods are very well done!

mervyn-teo commented:
Nicely done! The naming of classes and variable complied with the coding standard, comments/Javadoc are direct easy to understand!

mervyn-teo commented on src/main/java/thoughtbot/Parser.java:
```diff
@@ -0,0 +1,135 @@
+        String firstWord = userInput.split(" ")[0];
+
+        switch (firstWord) {
+        case "list":
```
Well organized switch cases that are easy to read.

mervyn-teo commented on src/main/java/thoughtbot/TaskList.java:
```diff
@@ -0,0 +1,196 @@
+     * @param t The task to be added
+     */
+    public void addTask(Task t) {
+        this.tasks.add(t);
```
Naming of methods are in camelCase, very nice!

mervyn-teo commented on src/main/java/thoughtbot/TaskList.java:
```diff
@@ -0,0 +1,196 @@
+ * The thoughtbot.TaskList object that keeps track of the Entry objects the user has currently added.
+ * This object also manages all the operations on the Entries itself.
+ */
+public class TaskList {
```
Naming of class are in Pascal Case, good!

mervyn-teo reviewed (commented):
Nicely done! The naming of classes and variable complied with the coding standard, comments/Javadoc are direct easy to understand!

deseansoh committed:
Add asserts to SaveLoad

These asserts are used to ensure that the code correctly throws
a LoadErrorException when the file is corrupted, and doesn't allow
the program to continue running with a corrupted save file.

deseansoh committed:
Add asserts to ThoughtBot

deseansoh committed:
Refactor Parser

This was done via Extract Function, to make it such that the main
Parser class looks cleaner, with just getter functions to get the
appropriate UserCommand object for each command.

Each getter function appropriately handles the logic and exception
handling for the Command it is associated with.

JavaDocs for all newly created methods are added as well.

deseansoh committed:
Merge pull request #1 from deseansoh/branch-A-Assertions

Branch Assertions

deseansoh committed:
Merge branch 'master' into branch-A-CodeQuality

deseansoh committed:
Merge pull request #2 from deseansoh/branch-A-CodeQuality

Branch Code Quality

deseansoh committed:
Add main.yml

deseansoh committed:
Remove IO redirection tests from main.yml

deseansoh committed:
Update codebase for checkstyle

deseansoh committed:
Add UserCommandRemind, Command.Remind, RemindFormat

deseansoh committed:
Add remind logic to Parser

deseansoh committed:
Add LocalDateTime getters to Deadline and Event

deseansoh committed:
Add remind logic to Tasklist and ThoughtBot

deseansoh committed:
Merge branch 'branch-B-Reminders'

deseansoh committed:
Edit to fulfill checkstyle

deseansoh committed:
Edit codebase for automated analysis bot

deseansoh committed:
Edit FX to show title

deseansoh committed:
Add Ui.png

deseansoh committed:
Update README.md

deseansoh committed:
Update README.md in GitHub page

deseansoh committed:
Add Ui.png to README.md
